adagrams/game_refactored.py

```diff
@@ -0,0 +1,209 @@
+    letter_scores = {}
+    point1 = (1, ["A", "E", "I", 'O', 'U', 'L', 'N', 'R', 'S', 'T'])
+    point2 = (2, ['D', 'G'])
+    point3 = (3, ['B', 'C', 'M', 'P']) 
+    point4 = (4, ['F', 'H', 'V', 'W', 'Y'])
+    point5 = (5, ['K']) 
+    point8 = (8, ['J', 'X'])
+    point10 = (10, ['Q', 'Z'])
+    point_info = [point1, point2, point3, point4, point5, point8, point10]
+    
+    for point_group in point_info:
+        for letter in point_group[1]:
+            letter_scores[letter] = point_group[0]
+            
+    return letter_scores
+    
+def get_score(word, letter_scores):
+    #loop through the word, adding up the value associated with the letter (key) of the dict!
+    score = 0
+    for char in word:
+        char = char.upper()
+        #look for the character in the keys of letter_scores.
+        #add the associated value to score
+        score += letter_scores[char]
+        #print(char, word, score)
+    if len(word) >= 7:
+        score += 8
+            
+    return score
+
+#--------------- New code from class --------------- 10/7/22 ---------
+#testing
+# letter_scores = make_score_dict()
+# print(get_score("za", letter_scores))
+
+# def max_with_ties_for_score(data):  #my function below should work for this as well. 
+#     best = []
+#     best_score = None
+#     for item in data:
+#         score = score_word(item)
+#         if best_score == None or score > best_score:
+#             best = [item] #throw out the old list!  start a new list.
+#             best_score = score
+#         elif score == best_score:
+#             best.append(item)
+#     return best
+
+#testing
+# words = ['hello', 'za', 'today', 'qi']
+# print(max_with_ties_for_score(words))
+
+# def max_with_ties_for_len(data):  #let's change a few things to make this about length. 
+#     best = []
+#     best_score = None
+#     for item in data:
+#         score = len(item)  #this is the only line that has changed so far!
+#         if best_score == None or score > best_score:
+#             best = [item] #throw out the old list!  start a new list.
+#             best_score = score
+#         elif score == best_score:
+#             best.append(item)
+#     return best
+
+#testing
+# words = ['hello', 'za', 'today', 'qi']
+# print(max_with_ties_for_len(words))
+
+#note : THE FUNCTIONS ABOVE ARE DUPLICATES OF EACH OTHER!!  TOO BUSY!!  le'ts get it together. 
+#let's replace this with higher order function!!
+#WE CAN USE THE FOLLOWING FUNCTION IN A LOT OF DIFFERENT CONTEXTS, JUST USING HIGHER ORDER FUNCTIONS.
+#this code is much more flexible. :) 
+
+#this behaves a lot like python's max.  Max is also flexible because it has an optional key argument.
+def max_with_ties(data, scorer):  #we'll pass in a function called scorer. #do all the logic with this function. 
+    best = []
+    best_score = None
+    for item in data:
+        score = scorer(item)  #this is the only line that has changed so far!
+        if best_score == None or score > best_score:
```

Comment:
As we covered in class, it's better convention to use `is` rather than `==` to check if a value is None!